FurEver/Pages/DogQuiz.cshtml.cs

```diff
@@ -0,0 +1,113 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace FurEver.Pages
+{
+    public class DogQuizModel : PageModel
+    {
+        public string QuizImageUrl { get; set; }
+        public List<string> BreedOptions { get; set; }
+        public string AnswerResult { get; set; }
+        public bool QuizFinished { get; set; } = false;
+
+        [TempData]
+        public int Score { get; set; }
+
+        [TempData]
+        public int CurrentQuestionNumber { get; set; } = 1;
+
+        [TempData]
+        public string CorrectBreed { get; set; }  // Store correct breed in TempData
+
+        public readonly int TotalQuestions = 5;
+
+        private static readonly string[] DogBreeds =
+        {
+            "beagle", "bulldog", "dalmatian", "goldenretriever", "labrador", "poodle", "pug", "shiba", "husky", "germanshepherd"
+        };
+
+        private static List<string> QuizImages { get; set; } = new List<string>();
+
+        public async Task OnGetAsync()
+        {
+            // Check if quiz is complete
+            if (CurrentQuestionNumber > TotalQuestions)
+            {
+                QuizFinished = true;
+                return;
```

Comment:
Some developers frown on mid-method returns.  The risk is, you might add needed logic below this, and not realize there's a path that would skip that logic.  Plus, most people expect to see returns at the end of a method.   A better practice is to use an if test, and have the return statement at the bottom of the method.